### DISCUSSION TITLE
Unreachable peer error when connecting, reading or writing to a siemens plc (s7-1200) using python-snap7 1.0

### DISCUSSION THREAD

**Discussion by iamtorchstack** [gijzelaerr/python-snap7#393] [Q&A] [answered]
Why do I intermittely get the  error
TCP : Unreachable peer
when connecting, reading or writing from a Siemens PLC siemens s7-1200.

I happen to have created an exception in the code to retry when it fails but I  desire to seek ways to eliminate this errors totally.

**Comment by swamper123** [comment]
Normally this indicates a network error. Check the port and address.
Can you ping your PLC? -> Then the IP stuff is okay.
Have you read [these notes](https://snap7.sourceforge.net/snap7_client.html#1200_1500) to avoid compatibility issues?

**Comment by gijzelaerr** [comment, marked as answer]
Closing as stale. This is a network configuration issue, not a library bug.
